src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -282,24 +336,37 @@ def _embed_system_under_purpose_or_feature(
     embedded_system_section.sort(key=lambda x: x.name)
 
 
-def build_purpose_or_feature_section_and_update_system_map(
+def build_purpose_or_feature_section_and_update_vendor_map(
     relevant_uses_or_features: List[str],
-    tcf_component_name: str,
-    system_map: Dict[str, TCFVendorRecord],
+    tcf_component_type: NonVendorSectionType,  # type: ignore[arg-type]
+    tcf_vendor_component_type: VendorSectionType,
+    vendor_subsection_name: str,
+    vendor_map: Dict[
+        str,
+        Union[
+            TCFConsentVendorRecord,
+            TCFLegitimateInterestsVendorRecord,
+            TCFVendorRelationships,
+        ],
```

Comment:
nit: don't you have this type defined above as `VendorRecord`? could you use that here?

Reply:
oh yes thank you!